src/createServer.js

```diff
@@ -1,3 +1,68 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+/* eslint-disable max-len */
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    res.setHeader('content-type', 'application/json');
+
+    const [url, queryString] = req.url.split('?');
+    const params = new URLSearchParams(queryString || '');
+
+    const text = url.replace(/^\//, '');
+    const toCase = params.get('toCase');
+
+    const errors = [];
+
+    if (!toCase) {
+      errors.push({
+        message: '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!text) {
+      errors.push({
+        message: 'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'].includes(toCase)) {
+      errors.push({
+        message: 'This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+      });
+    }
+
+    if (errors.length > 0) {
+      res.writeHead(400, 'Bad Request');
+      res.end(JSON.stringify({ errors }));
+
+      return;
+    }
+
+    const result = convertToCase(text, toCase);
+
+    if (!result || !result.originalCase || !result.convertedText) {
```

Comment:
its better to use destructuring
`const { originalCase, convertedText } = convertToCase(text, toCase);`